ql/src/java/org/apache/hadoop/hive/ql/exec/repl/bootstrap/ReplLoadTask.java

```diff
@@ -153,6 +165,13 @@ a database ( directory )
           partitionsPostProcessing(iterator, scope, loadTaskTracker, tableTracker,
               partitionsTracker);
           partitionsTracker.debugLog("partitions");
+
+          if (!loadTaskTracker.hasReplicationState()) {
+            // Add ReplStateLogTask only if no pending table load tasks left for next cycle
+            ImportTableDesc tableDesc = loadPartitions.tableDesc();
+            loadTaskTracker.updateTaskCount(createTableReplLogTask(partitionsTracker.tasks(),
```

Comment:
It is not as table DAG was added separately and ReplLoadtask was added later. When I move it inside LoadTable/Loadpartitions, probably this will be done implicitly.